server.go

```diff
@@ -0,0 +1,129 @@
+package snmpgo
+
+import (
+	"log"
+	"net"
+	"runtime"
+	"time"
+)
+
+const (
+	maxTrapSize = 2 << 11 // 2048 bytes
+)
+
+// TrapListener defines method that need to be implemented by Trap listeners.
+// If OnTRAP panics, the server (caller of OnTRAP) assumes that affect of the panic
+// is temporary and recovers by the panic and logs trace to the error log.
+type TrapListener interface {
+	OnTRAP(trap *TrapRequest)
+}
+
+/**
+ *  TrapRequest is representing trap request that is send from the network element.
+ */
+type TrapRequest struct {
+	// The received Trap message
+	Message message
```

Comment:
`TrapRequest` structure has `message` member, but this is private interface.
Is it possible to substitute with `Pdu`?